specification/sections/02-providers.md

```diff
@@ -26,6 +26,22 @@ Providers are the "translator" between the flag evaluation calls made in applica
 provider.getMetadata().getName(); // "my-custom-provider"
 ```
 
+#### Requirement 2.1.2
+
+> The provider `metadata` member or accessor **MAY** define an `allFlagKeys` field or accessor of type collection of strings, which identifies all available flag keys in the provider.
+
+```typescript
+provider.getMetadata().getAllFlagKeys(); // ['featureA', 'featureB']
+```
+
+#### Condition 2.1.3
+
+> The implementing language has an asynchronous support.
+
+##### Conditional Requirement 2.1.3.1
+
+> The accessor for `allFlagKeys` **SHOULD** be asynchronous.
+
```

Comment:
I am unsure if we need to call this out, but it felt important, given how expensive this call might be. I'm happy to remove this if it's overly restrictive.

Reply:
I think it's a good idea to call out. Makes it more obvious and clear imho